SSA/Projects/Tensor2D/Tensor2D.lean

```diff
@@ -0,0 +1,225 @@
+import SSA.Core.WellTypedFramework
+import Mathlib.Data.Matrix.Basic
+import Mathlib.Data.Nat.Basic
+import SSA.Core.Util
+
+
+/-- Tensor2d with existential dimension sizes. -/
+structure Tensor2d' (α : Type) where
+  n₀ : ℕ
+  n₁ : ℕ
+  mat : Matrix (Fin n₀) (Fin n₁) α
+
+def Tensor2d'.error (α : Type) : Tensor2d' α where
```

Comment:
do you think this is more elegant than using `Option`. It would break reasoning about the dimensions being preserved, for example, in some cases. I like the explicit option more because it allows you to separate the error mode cleanly